lib/types.ts

```diff
@@ -231,9 +231,9 @@ export interface EntryAPI extends TaskAPI {
    * Optional metadata on an entry
    * @deprecated
    */
-  metadata?: Metadata,
-  getMetadata: () => Metadata,
-  onMetadataChanged: (callback: (metadata: Metadata) => void) => Function,
+  metadata?: Metadata | undefined
```

Comment:
nit: `metadata?: Metadata` is actually the same as `metadata?: Metadata | undefined`